src/core/searcher.rs

```diff
@@ -247,6 +247,14 @@ impl SearcherInner {
         generation: TrackedObject<SearcherGeneration>,
         doc_store_cache_size: usize,
     ) -> io::Result<SearcherInner> {
+        debug_assert_eq!(
```

Comment:
As discussed in the other thread, I prefer having an actually assert here. (as opposed to debug_assert)